forum/profiles/permissions.py

```diff
@@ -22,3 +25,24 @@ def has_object_permission(self, request, view, obj):
         # Write permissions are only allowed to the owner of the object
         return obj.user == request.user
 
+
+class AbstractRolePermission(BasePermission):
+    role = None
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        try:
+            token_role_value = request.auth.get('role')
+        except AttributeError:
+            raise AuthenticationFailed("Invalid token")
+
+        return int(token_role_value) == self.role
```

Comment:
The use of `int(token_role_value)` assumes that `token_role_value` can always be cast to an integer. If not, it could raise a `ValueError`.